# 파이썬 시작하기

001 print 기초
화면에 Hello World 문자열을 출력하세요.

In [1]:
'hello world'

'hello world'

002 print 기초
화면에 Mary's cosmetics을 출력하세요. (중간에 '가 있음에 주의하세요)

In [6]:
"Mary's cosmetics"

"Mary's cosmetics"

003 print 기초
화면에 아래 문장을 출력하세요. (중간에 "가 있음에 주의하세요.)

In [12]:
print('신씨가 소리질렀다. "도둑이야"')

신씨가 소리질렀다. "도둑이야"


004 print 기초
화면에 C:\Windows를 출력하세요.

In [34]:
print('C:\Windows', 'C:/windows') # / 도 됨

C:\Windows C:/windows


005 print 탭과 줄바꿈
다음 코드를 실행해보고 \t와 \n의 역할을 설명해보세요.

In [16]:
print("안녕하세요.\n만나서\t\t반갑습니다.")

# \n : 줄바꿈 \t : 탭삽입

안녕하세요.
만나서		반갑습니다.


006 print 여러 데이터 출력
print 함수에 두 개의 단어를 입력한 예제입니다. 아래 코드의 출력 결과를 예상해봅시다.

In [32]:
print("오늘은", "일요일")

오늘은 일요일


007 print 기초
print() 함수를 사용하여 다음과 같이 출력하세요.

In [39]:
print('naver', 'kakao', 'sk', 'samsung', sep = ';')
print(";".join(['naver', 'kakao', 'sk', 'samsung'])) # 조인도 됨

naver;kakao;sk;samsung
naver;kakao;sk;samsung


008 print 기초
print() 함수를 사용하여 다음과 같이 출력하세요.

In [25]:
print('naver', 'kakao', 'sk', 'samsung', sep = '/')

naver/kakao/sk/samsung


009 print 줄바꿈
다음 코드를 수정하여 줄바꿈이 없이 출력하세요. (힌트: end='') print 함수는 두 번 사용합니다. 세미콜론 (;)은 한줄에 여러 개의 명령을 작성하기 위해 사용합니다.

In [43]:
print("fisrt", end="");print("second")

fisrtsecond


010 연산 결과 출력
5/3의 결과를 화면에 출력하세요.


In [45]:
print(5/3)

1.6666666666666667


011 변수 사용하기
삼성전자라는 변수로 50,000원을 바인딩해보세요. 삼성전자 주식 10주를 보유하고 있을 때 총 평가금액을 출력하세요.

In [47]:
삼성전자 = 50000
총평가금액 = 삼성전자 * 10
print(총평가금액)

500000


012 변수 사용하기
다음 표는 삼성전자의 일부 투자정보입니다. 변수를 사용해서 시가총액, 현재가, PER 등을 바인딩해보세요.

항목	값
시가총액	298조
현재가	50,000원
PER	15.79

교안에서는 이렇게 했음

시가총액 = 298000000000000
현재가 = 50000
PER = 15.79
print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))

* 바인딩이 변수랑 타입까지 출력하라는거인듯?

In [54]:
시가총액 = '298조'
현재가 = '50,000원'
PER = 15.79

print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))
      

298조 <class 'str'>
50,000원 <class 'str'>
15.79 <class 'float'>


013 문자열 출력
변수 s와 t에는 각각 문자열이 바인딩 되어있습니다.

>> s = "hello"
>> t = "python"
두 변수를 이용하여 아래와 같이 출력해보세요.

실행 예:
hello! python

In [63]:
s = 'hello'
t = 'python'

print(s + '! ' + t)

hello! python


014 파이썬을 이용한 값 계산
아래 코드의 실행 결과를 예상해보세요.

>> 2 + 2 * 3 

8

In [64]:
2 + 2 * 3 

8

015 type 함수
type() 함수는 데이터 타입을 판별합니다. 변수 a에는 128 숫자가 바인딩돼 있어 type 함수가 int (정수)형임을 알려줍니다.

>> a = 128
>> print (type(a))
<class 'int'>
아래 변수에 바인딩된 값의 타입을 판별해보세요.

>> a = "132"

In [2]:
a = 128
print(type(a))

a = "132"
print(type(a))

<class 'int'>
<class 'str'>


016 문자열을 정수로 변환
문자열 '720'를 정수형으로 변환해보세요.

>> num_str = "720"

In [10]:
num_str = '720'
type(num_str)

num_str = int(num_str)
print(type(num_str))


<class 'int'>


017 정수를 문자열 100으로 변환
정수 100을 문자열 '100'으로 변환해보세요.

In [12]:
num_int2str = 100
num_str2int = '100'

num_int2str = str(num_int2str)
num_str2int = int(num_str2int)


print(num_int2str, num_str2int, type(num_int2str), type(num_str2int))

100 100 <class 'str'> <class 'int'>


018 문자열을 실수로 변환
문자열 "15.79"를 실수(float) 타입으로 변환해보세요.

In [13]:
a = '15.79'
result = float(a)
print(type(result))

<class 'float'>


019 문자열을 정수로 변환
year라는 변수가 문자열 타입의 연도를 바인딩하고 있습니다. 이를 정수로 변환한 후 최근 3년의 연도를 화면에 출력해보세요.

year = "2020"

In [14]:
year = '2020'
result = int(year)
print(result, result-1, result-2)

2020 2019 2018


020 파이썬 계산
에이컨이 월 48,584원에 무이자 36개월의 조건으로 홈쇼핑에서 판매되고 있습니다. 총 금액은 계산한 후 이를 화면에 출력해보세요. (변수사용하기)

In [15]:
월요금 = 48584
개월수 = 36

총금액 = 월요금 * 개월수

print(총금액)

1749024


021 문자열 인덱싱
letters가 바인딩하는 문자열에서 첫번째와 세번째 문자를 출력하세요.

>> letters = 'python'
실행 예
p t

In [18]:
letters = 'python'

print(letters[0], letters[2])

p t


022 문자열 슬라이싱
자동차 번호가 다음과 같을 때 뒤에 4자리만 출력하세요.

>> license_plate = "24가 2210"
실행 예: 2210

In [22]:
license_plate = '24가 2210'

license_plate[-4:]

'2210'

023 문자열 인덱싱
아래의 문자열에서 '홀' 만 출력하세요.

>> string = "홀짝홀짝홀짝"
실행 예:
홀홀홀

In [24]:
string = '홀짝홀짝홀짝'

print(string[0], string[2], string[4])

# 교안에서는 print(string[::2])

홀 홀 홀


024 문자열 슬라이싱
문자열을 거꾸로 뒤집어 출력하세요.

>> string = "PYTHON"
실행 예:
NOHTYP

In [31]:
string = 'python'

print(string[::-1])

nohtyp


025 문자열 치환
아래의 전화번호에서 하이푼 ('-')을 제거하고 출력하세요.

>> phone_number = "010-1111-2222"
실행 예
010 1111 2222

In [32]:
phone_number = '010-1111-2222'

phone_number.replace('-',' ')

'010 1111 2222'

026 문자열 다루기
25번 문제의 전화번호를 아래와 같이 모두 붙여 출력하세요.

실행 예
01011112222

In [34]:
phone_number.replace('-','')

'01011112222'

027 문자열 다루기
url 에 저장된 웹 페이지 주소에서 도메인을 출력하세요.

>> url = "http://sharebook.kr"
실행 예:
kr

In [36]:
url = "http://sharebook.kr"
url[-2:]

# 교안
# url_split = url.split('.')
# print(url_split[-1])

'kr'

028 문자열은 immutable
아래 코드의 실행 결과를 예상해보세요.

>> lang = 'python'
>> lang[0] = 'P'
>> print(lang)

In [ ]:
2번행에서 오류 발생

029 replace 메서드
아래 문자열에서 소문자 'a'를 대문자 'A'로 변경하세요.

>> string = 'abcdfe2a354a32a'
실행 예:
Abcdfe2A354A32A

In [37]:
string = 'abcdfe2a354a32a'

string.replace('a', 'A')

'Abcdfe2A354A32A'

030 replace 메서드
아래 코드의 실행 결과를 예상해보세요.

>> string = 'abcd'
>> string.replace('b', 'B')
>> print(string)

In [ ]:
abcd

031 문자열 합치기
아래 코드의 실행 결과를 예상해보세요.

>> a = "3"
>> b = "4"
>> print(a + b)

In [ ]:
34

032 문자열 곱하기
아래 코드의 실행 결과를 예상해보세요.

>> print("Hi" * 3)

In [ ]:
HiHiHi

033 문자열 곱하기
화면에 '-'를 80개 출력하세요.

실행 예:
--------------------------------------------------------------------------------

In [38]:
print('-'*80)

--------------------------------------------------------------------------------


034 문자열 곱하기
변수에 다음과 같은 문자열이 바인딩되어 있습니다.

>>> t1 = 'python'
>>> t2 = 'java'
변수에 문자열 더하기와 문자열 곱하기를 사용해서 아래와 같이 출력해보세요.

실행 예:
python java python java python java python java

In [40]:
t1 = 'python'
t2 = 'java'

print((t1+' '+t2)*4)

python javapython javapython javapython java


035 문자열 출력
변수에 다음과 같이 문자열과 정수가 바인딩되어 있을 때 % formatting을 사용해서 다음과 같이 출력해보세요.

name1 = "김민수" 
age1 = 10
name2 = "이철희"
age2 = 13
이름: 김민수 나이: 10
이름: 이철희 나이: 13

In [48]:
name1 = '김민수'
age1 = 10
name2 = '이철희'
age2 = 13

print('이름: {}, 나이: {}'.format(name1, age1))
print('이름: {}, 나이: {}'.format(name2, age2))

#교안에선 이렇게 풀었음
#print("이름: %s 나이: %d" % (name1, age1))
#print("이름: %s 나이: %d" % (name2, age2))


이름: 김민수, 나이: 10
이름: 이철희, 나이: 13


036 문자열 출력
문자열의 format( ) 메서드를 사용해서 035번 문제를 다시 풀어보세요.

In [ ]:
# 위에서가 %s %d 로 하고 여기서 {}로 풀라는거 같음

037 문자열 출력
파이썬 3.6부터 지원하는 f-string을 사용해서 035번 문제를 다시 풀어보세요.

In [51]:
print(f'이름: {name1}, 나이: {age1}')
print(f'이름: {name2}, 나이: {age2}')

이름: 김민수, 나이: 10
이름: 이철희, 나이: 13


038 컴마 제거하기
삼성전자의 상장주식수가 다음과 같습니다. 컴마를 제거한 후 이를 정수 타입으로 변환해보세요.

상장주식수 = "5,969,782,550"

In [61]:
상장주식수 = "5,969,782,550"

상장주식수 = 상장주식수.replace(",","")
상장주식수 = int(상장주식수)
print(상장주식수, type(상장주식수))

5969782550 <class 'int'>


039 문자열 슬라이싱
다음과 같은 문자열에서 '2020/03'만 출력하세요.

분기 = "2020/03(E) (IFRS연결)"

In [64]:
분기 = "2020/03(E) (IFRS연결)"
분기[0:7]

'2020/03'

040 strip 메서드
문자열의 좌우의 공백이 있을 때 이를 제거해보세요.

data = "   삼성전자    "

In [70]:
data = "   삼성전자    "
result = data.strip()
result

'삼성전자'

041 upper 메서드
다음과 같은 문자열이 있을 때 이를 대문자 BTC_KRW로 변경하세요.

ticker = "btc_krw"

In [72]:
ticker = "btc_krw"
result = ticker.upper()
result

'BTC_KRW'

042 lower 메서드
다음과 같은 문자열이 있을 때 이를 소문자 btc_krw로 변경하세요.

ticker = "BTC_KRW"

In [74]:
result2 = ticker.lower()
result2

'btc_krw'

043 capitalize 메서드
문자열 'hello'가 있을 때 이를 'Hello'로 변경해보세요.

In [76]:
a = 'hello'
result = a.capitalize()
result

'Hello'

044 endswith 메서드
파일 이름이 문자열로 저장되어 있을 때 endswith 메서드를 사용해서 파일 이름이 'xlsx'로 끝나는지 확인해보세요.

file_name = "보고서.xlsx"

In [79]:
file_name = "보고서.xlsx"

file_name.endswith('xlsx')

True

In [ ]:
046 startswith 메서드
파일 이름이 문자열로 저장되어 있을 때 startswith 메서드를 사용해서 파일 이름이 '2020'로 시작하는지 확인해보세요.

file_name = "2020_보고서.xlsx"

In [80]:
file_name = "2020_보고서.xlsx"
file_name.startswith('2020')

True

047 split 메서드
다음과 같은 문자열이 있을 때 공백을 기준으로 문자열을 나눠보세요.

a = "hello world"

In [84]:
a = "hello world"
a.split()

['hello', 'world']

048 split 메서드
다음과 같이 문자열이 있을 때 btc와 krw로 나눠보세요.

ticker = "btc_krw"

In [85]:
ticker = "btc_krw"

ticker.split('_')

['btc', 'krw']

049 split 메서드
다음과 같이 날짜를 표현하는 문자열이 있을 때 연도, 월, 일로 나눠보세요.

date = "2020-05-01"

In [86]:
date = "2020-05-01"
date.split('-')

['2020', '05', '01']

050 rstrip 메서드
문자열의 오른쪽에 공백이 있을 때 이를 제거해보세요.

data = "039490     "

In [87]:
data = "039490     "
data.rstrip()


'039490'

051 리스트 생성
2016년 11월 영화 예매 순위 기준 top3는 다음과 같습니다. 영화 제목을 movie_rank 이름의 리스트에 저장해보세요. (순위 정보는 저장하지 않습니다.)

순위	영화
1	닥터 스트레인지
2	스플릿
3	럭키

In [90]:
movie_rank = ['닥터 스트레인지', '스플릿', '럭키']
movie_rank

['닥터 스트레인지', '스플릿', '럭키']

052 리스트에 원소 추가
051의 movie_rank 리스트에 "배트맨"을 추가하라.

In [92]:
movie_rank = movie_rank + ['배트맨']
movie_rank

# 교안에선 이렇게 풀었음
# movie_rank = ["닥터 스트레인지", "스플릿", "럭키"]
# movie_rank.append("배트맨")
# print(movie_rank)

['닥터 스트레인지', '스플릿', '럭키', '배트맨']

053
movie_rank 리스트에는 아래와 같이 네 개의 영화 제목이 바인딩되어 있다. "슈퍼맨"을 "닥터 스트레인지"와 "스플릿" 사이에 추가하라.

movie_rank = ['닥터 스트레인지', '스플릿', '럭키', '배트맨']

In [94]:
movie_rank = ['닥터 스트레인지', '스플릿', '럭키', '배트맨']
movie_rank.insert(1, '슈퍼맨')
movie_rank

['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']

054
movie_rank 리스트에서 '럭키'를 삭제하라.

movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']

In [98]:
movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']
movie_rank.remove('럭키')
movie_rank


# movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']
# del movie_rank[3]
# print(movie_rank)

['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']

055
movie_rank 리스트에서 '스플릿' 과 '배트맨'을 를 삭제하라.

movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']

In [103]:
movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']
movie_rank.remove('스플릿')
movie_rank.remove('배트맨')
movie_rank

#movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']
#del movie_rank[2]
#del movie_rank[2]
#print(movie_rank)

['닥터 스트레인지', '슈퍼맨']

056
lang1과 lang2 리스트가 있을 때 lang1과 lang2의 원소를 모두 갖고 있는 langs 리스트를 만들어라.

>> lang1 = ["C", "C++", "JAVA"]
>> lang2 = ["Python", "Go", "C#"]
실행 예:
>> langs
['C', 'C++', 'JAVA', 'Python', 'Go', 'C#']

In [104]:
lang1 = ["C", "C++", "JAVA"]
lang2 = ["Python", "Go", "C#"]

langs = lang1 + lang2
langs

['C', 'C++', 'JAVA', 'Python', 'Go', 'C#']

057
다음 리스트에서 최댓값과 최솟값을 출력하라. (힌트: min(), max() 함수 사용)

nums = [1, 2, 3, 4, 5, 6, 7]
실행 예:
max:  7
min:  1

In [116]:
nums = [1, 2, 3, 4, 5, 6, 7]

print('max: ', max(nums))
print('min: ', min(nums))

max:  7
min:  1


058
다음 리스트의 합을 출력하라.

nums = [1, 2, 3, 4, 5]
실행 예:
15

In [118]:
nums = [1, 2, 3, 4, 5]
sum(nums)

15

059
다음 리스트에 저장된 데이터의 개수를 화면에 구하하라.

cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]

In [124]:
cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]

len(cook)

12

060
다음 리스트의 평균을 출력하라.

nums = [1, 2, 3, 4, 5]
실행 예:
3.0

In [130]:
nums = [1, 2, 3, 4, 5]

sum(nums) / len(nums)

3.0

061
price 변수에는 날짜와 종가 정보가 저장돼 있다. 날짜 정보를 제외하고 가격 정보만을 출력하라. (힌트 : 슬라이싱)

price = ['20180728', 100, 130, 140, 150, 160, 170]
출력 예시:
[100, 130, 140, 150, 160, 170]

In [132]:
price = ['20180728', 100, 130, 140, 150, 160, 170]

price[1:]

[100, 130, 140, 150, 160, 170]

062
슬라이싱을 사용해서 홀수만 출력하라.

nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
실행 예:
[1, 3, 5, 7, 9]

In [138]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
nums[::2]

[1, 3, 5, 7, 9]

슬라이싱을 사용해서 짝수만 출력하라.

nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
실행 예:
[2, 4, 6, 8, 10]

In [141]:
nums[1::2]

[2, 4, 6, 8, 10]

064
슬라이싱을 사용해서 리스트의 숫자를 역 방향으로 출력하라.

nums = [1, 2, 3, 4, 5]
실행 예:
[5, 4, 3, 2, 1]

In [144]:
nums = [1, 2, 3, 4, 5]

nums[::-1]

[5, 4, 3, 2, 1]

065
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver']
interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자 Naver

In [145]:
interest = ['삼성전자', 'LG전자', 'Naver']
interest[::2]

#print(interest[0], interest[2])

['삼성전자', 'Naver']

066 join 메서드
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자 LG전자 Naver SK하이닉스 미래에셋대우

In [147]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
" ".join(interest)

'삼성전자 LG전자 Naver SK하이닉스 미래에셋대우'

067 join 메서드
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우

In [148]:
"/".join(interest)

'삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우'

068 join 메서드
interest 리스트에는 아래의 데이터가 바인딩되어 있다.

interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
join() 메서드를 사용해서 interest 리스트를 아래와 같이 화면에 출력하라.

출력 예시:
삼성전자
LG전자
Naver
SK하이닉스
미래에셋대우

In [153]:
print('\n'.join(interest))

삼성전자
LG전자
Naver
SK하이닉스
미래에셋대우


069 문자열 split 메서드
회사 이름이 슬래시 ('/')로 구분되어 하나의 문자열로 저장되어 있다.

string = "삼성전자/LG전자/Naver"
이를 interest 이름의 리스트로 분리 저장하라.

실행 예시
>> print(interest)
['삼성전자', 'LG전자', 'Naver']

In [158]:
string = "삼성전자/LG전자/Naver"

interest = string.split('/')
print(interest)

['삼성전자', 'LG전자', 'Naver']


070 리스트 정렬
리스트에 있는 값을 오름차순으로 정렬하세요.

data = [2, 4, 3, 1, 5, 10, 9]

In [166]:
data = [2, 4, 3, 1, 5, 10, 9]
data.sort()
data

[1, 2, 3, 4, 5, 9, 10]

071
my_variable 이름의 비어있는 튜플을 만들라.

In [171]:
my_variable = ()
print(my_variable, type(my_variable))


() <class 'tuple'>


072
2016년 11월 영화 예매 순위 기준 top3는 다음과 같다. 영화 제목을 movie_rank 이름의 튜플에 저장하라. (순위 정보는 저장하지 않는다.)

순위	영화
1	닥터 스트레인지
2	스플릿
3	럭키

In [172]:
movie_rank = ('닥터 스트레인지', '스플릿', '럭키')
movie_rank

('닥터 스트레인지', '스플릿', '럭키')

073
숫자 1 이 저장된 튜플을 생성하라.

In [175]:
a = (1)
b = (1, )
print(a, type(a), b, type(b))

1 <class 'int'> (1,) <class 'tuple'>


074
다음 코드를 실행해보고 오류가 발생하는 원인을 설명하라.

>> t = (1, 2, 3)
>> t[0] = 'a'
Traceback (most recent call last):
  File "<pyshell#46>", line 1, in <module>
    t[0] = 'a'
TypeError: 'tuple' object does not support item assignment

In [179]:
# 튜플은 수정이 안됨

075
아래와 같이 t에는 1, 2, 3, 4 데이터가 바인딩되어 있다. t가 바인딩하는 데이터 타입은 무엇인가?

t = 1, 2, 3, 4

In [180]:
t = 1, 2, 3, 4

type(t)

# 괄호가 없어도 튜플로 동작함

tuple

076
변수 t에는 아래와 같은 값이 저장되어 있다. 변수 t가 ('A', 'b', 'c') 튜플을 가리키도록 수정 하라.

t = ('a', 'b', 'c')

In [183]:
t = ('a', 'b', 'c')
t = ('A', 'b', 'c')
# 튜플은 새로 만들어야 한다

077
다음 튜플을 리스트로 변환하라.

interest = ('삼성전자', 'LG전자', 'SK Hynix')

In [188]:
interest = ('삼성전자', 'LG전자', 'SK Hynix')
interest = ['삼성전자', 'LG전자', 'SK Hynix']
interest

# data = list(interest)
# data

['삼성전자', 'LG전자', 'SK Hynix']

078
다음 리스트를 튜플로 변경하라.

interest = ['삼성전자', 'LG전자', 'SK Hynix']

In [187]:
interest = ['삼성전자', 'LG전자', 'SK Hynix']
data = tuple(interest)
data

('삼성전자', 'LG전자', 'SK Hynix')

079 튜플 언팩킹
다음 코드의 실행 결과를 예상하라.

temp = ('apple', 'banana', 'cake')
a, b, c = temp
print(a, b, c)

In [191]:
temp = ('apple', 'banana', 'cake')
a, b, c = temp
print(a, b, c)

# apple banana cake

apple banana cake


080 range 함수
1 부터 99까지의 정수 중 짝수만 저장된 튜플을 생성하라.

(2, 4, 6, 8 ... 98)

In [204]:
data = tuple(range(2, 99, 2))
data

(2,
 4,
 6,
 8,
 10,
 12,
 14,
 16,
 18,
 20,
 22,
 24,
 26,
 28,
 30,
 32,
 34,
 36,
 38,
 40,
 42,
 44,
 46,
 48,
 50,
 52,
 54,
 56,
 58,
 60,
 62,
 64,
 66,
 68,
 70,
 72,
 74,
 76,
 78,
 80,
 82,
 84,
 86,
 88,
 90,
 92,
 94,
 96,
 98)

In [ ]:
081 별 표현식
기본적으로 데이터 언패킹은 좌변의 변수와 우변 데이터 개수가 같아야 합니다. 하지만 star expression을 사용하면 변수의 개수가 달라도 데이터 언패킹을 할 수 있습니다. 튜플에 저장된 데이터 중에서 앞에 있는 두 개의 데이터만 필요할 경우 나머지 데이터의 언패킹 코드를 작성할 필요가 없습니다.

>> a, b, *c = (0, 1, 2, 3, 4, 5)
>> a
0
>> b
1
>> c
[2, 3, 4, 5]
다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 좌측 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [1]:
a, b, *c = (0, 1, 2, 3, 4, 5)
a

0

In [2]:
b

1

In [3]:
c

[2, 3, 4, 5]

In [48]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

*valid_score, a, b = scores
valid_score

[8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5]
9.4
7.8


082
다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 우측 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [9]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

a, b, *valid_score = scores
valid_score

[8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

083
다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 가운데 있는 8개의 값을 valid_score 변수에 바인딩하여라.

scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [10]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

a, *valid_score, b = scores
valid_score

[8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8]

084 비어있는 딕셔너리
temp 이름의 비어있는 딕셔너리를 만들라.

In [13]:
temp = { }
temp

{}

085
다음 아이스크림 이름과 희망 가격을 딕셔너리로 구성하라.

이름	희망 가격
메로나	1000
폴라포	1200
빵빠레	1800

In [24]:
dic = {'메로나' : 1000, '폴라포' : 1200, '빵빠레' : 1800}
dic

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800}

085 번의 딕셔너리에 아래 아이스크림 가격정보를 추가하라.

이름	희망 가격
죠스바	1200
월드콘	1500

In [26]:
dic['죠스바'] = 1200
dic['월드콘'] = 1500
print(dic)

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


087
다음 딕셔너리를 사용하여 메로나 가격을 출력하라.

ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}
실행 예:
메로나 가격: 1000

In [40]:
ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

print('메로나 가격:', ice['메로나'])


메로나 가격: 1000


088
다음 딕셔너리에서 메로나의 가격을 1300으로 수정하라.

ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

In [41]:
ice['메로나'] = 1300
ice

{'메로나': 1300, '폴로포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}

089
다음 딕셔너리에서 메로나를 삭제하라.

ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

In [45]:
ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

del ice['메로나']
ice

{'폴로포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}

090
다음 코드에서 에러가 발생한 원인을 설명하라.

>> icecream = {'폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
>> icecream['누가바']
Traceback (most recent call last):
  File "<pyshell#69>", line 1, in <module>
    icecream['누가바']
KeyError: '누가바'

In [ ]:
icecream 딕셔너리에 누가바 키가 없음

091 딕셔너리 생성
아래의 표에서, 아이스크림 이름을 키값으로, (가격, 재고) 리스트를 딕셔너리의 값으로 저장하라. 딕셔너리의 이름은 inventory로 한다.

이름	가격	재고
메로나	300	20
비비빅	400	3
죠스바	250	100

In [49]:
inventory = {'메로나' : [300, 20], '비비빅' : [400, 3], '죠스바' : [250, 100]}
inventory

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100]}

092 딕셔너리 인덱싱
inventory 딕셔너리에서 메로나의 가격을 화면에 출력하라.

inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}
실행 예시:
300 원

In [57]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}

print(inventory["메로나"][0], '원')

300 원


093 딕셔너리 인덱싱
inventory 딕셔너리에서 메로나의 재고를 화면에 출력하라.

inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}
실행 예시:
20 개

In [63]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}

print(inventory["메로나"][0], "개")

300 개


094 딕셔너리 추가
inventory 딕셔너리에 아래 데이터를 추가하라.

inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}
이름	가격	재고
월드콘	500	7
실행 예시:
>> print(inventory)
{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100], '월드콘': [500, 7]}

In [68]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}
inventory["월드콘"] = [500, 7]
inventory

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100], '월드콘': [500, 7]}

095 딕셔너리 keys() 메서드
다음의 딕셔너리로부터 key 값으로만 구성된 리스트를 생성하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [73]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

list(icecream.keys())

['탱크보이', '폴라포', '빵빠레', '월드콘', '메로나']

097 딕셔너리 values() 메서드
icecream 딕셔너리에서 아이스크림 판매 금액의 총합을 출력하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
출력 예시:
6700

In [72]:
sum(icecream.values())

6700

098 딕셔너리 update 메서드
아래의 new_product 딕셔너리를 다음 icecream 딕셔너리에 추가하라.

icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
new_product = {'팥빙수':2700, '아맛나':1000}
실행 예시:
>> print(icecream)
{'탱크보이': 1200,  '폴라포': 1200,  '빵빠레': 1800,  '월드콘': 1500,  '메로나': 1000,  '팥빙수':2700, '아맛나':1000}

In [ ]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
new_product = {'팥빙수':2700, '아맛나':1000}
    
    

099 zip과 dict
아래 두 개의 튜플을 하나의 딕셔너리로 변환하라. keys를 키로, vals를 값으로 result 이름의 딕셔너리로 저장한다.

keys = ("apple", "pear", "peach")
vals = (300, 250, 400)
실행 예시:
>> print(result)
{'apple': 300, 'pear': 250, 'peach': 400}

In [83]:
keys = ("apple", "pear", "peach")
vals = (300, 250, 400)

result = dict(zip(keys, vals))
result

{'apple': 300, 'pear': 250, 'peach': 400}

100 zip과 dict
date와 close_price 두 개의 리스트를 close_table 이름의 딕셔너리로 생성하라.

date = ['09/05', '09/06', '09/07', '09/08', '09/09']
close_price = [10500, 10300, 10100, 10800, 11000]
실행 예시:
>> print(close_table)
{'09/05': 10500, '09/06': 10300, '09/07': 10100, '09/08': 10800, '09/09': 11000}

In [85]:
date = ['09/05', '09/06', '09/07', '09/08', '09/09']
close_price = [10500, 10300, 10100, 10800, 11000]
close_table = dict(zip(date, close_price))
close_table

{'09/05': 10500,
 '09/06': 10300,
 '09/07': 10100,
 '09/08': 10800,
 '09/09': 11000}

101
파이썬에서 True 혹은 False를 갖는 데이터 타입은 무엇인가?

In [ ]:
bool 타입

102
아래 코드의 출력 결과를 예상하라

print(3 == 5)

In [ ]:
False

103
아래 코드의 출력 결과를 예상하라

print(3 < 5)

In [ ]:
True

104
아래 코드의 결과를 예상하라.

x = 4
print(1 < x < 5)

In [ ]:
True

105
아래 코드의 결과를 예상하라.

print ((3 == 3) and (4 != 3))

In [ ]:
# true and true

True

106
아래 코드에서 에러가 발생하는 원인에 대해 설명하라.

print(3 => 4)

In [ ]:
부등호 순서가 잘못됨

107
아래 코드의 출력 결과를 예상하라

if 4 < 3:
    print("Hello World")

In [ ]:
조건문이 false 여서 출력이 안됨

108
아래 코드의 출력 결과를 예상하라

if 4 < 3:
    print("Hello World.")
else:
    print("Hi, there.")

In [ ]:
조건문이 false여서 Hi, there 가 출력됨

109
아래 코드의 출력 결과를 예상하라

if True :
    print ("1")
    print ("2")
else :
    print("3")
print("4")

In [ ]:
조건문이 true 이고 조건문 밖에 print('4')가 있기 때문에

1
2
4

가 출력 됨

110
아래 코드의 출력 결과를 예상하라

if True :
    if False:
        print("1")
        print("2")
    else:
        print("3")
else :
    print("4")
print("5")

In [1]:
# 외부 if 문에서 True 이고 내부 if문에서 조건이 false 이기 때문에
# else로 넘어가서 '3'이 프린트 되고 외부 if문을 벗어나서 '5'가 추가로 출력됨

3
5

5

111
사용자로부터 입력받은 문자열을 두 번 출력하라. 아래는 사용자가 "안녕하세요"를 입력한 경우의 출력 결과이다.

>> 안녕하세요
안녕하세요안녕하세요

In [3]:
txt = input()
print(txt * 2)

안녕하세요
안녕하세요안녕하세요


112
사용자로부터 하나의 숫자를 입력받고, 입력 받은 숫자에 10을 더해 출력하라.

>> 숫자를 입력하세요: 30
40

In [3]:
num = input()
result = int(num) + 10
print(result)

30
40


113
사용자로부터 하나의 숫자를 입력 받고 짝수/홀수를 판별하라.

>> 30
짝수

In [10]:
num = input()
if int(num) % 2 == 0 :
    print('짝수')
else :
    print('홀수')

2
짝수


114
사용자로부터 값을 입력받은 후 해당 값에 20을 더한 값을 출력하라. 단 사용자가 입력한 값과 20을 더한 계산 값이 255를 초과하는 경우 255를 출력해야 한다.

>> 입력값: 200
출력값: 220
>> 입력값: 240
출력값: 255

In [21]:
num = input()
num = int(num) + 20
if num > 255 :
    print(255)
else :
    print(num)


245
255


115
사용자로부터 하나의 값을 입력받은 후 해당 값에 20을 뺀 값을 출력하라. 단 출력 값의 범위는 0~255이다. 예를 들어 결괏값이 0보다 작은 값이되는 경우 0을 출력하고 255보다 큰 값이 되는 경우 255를 출력해야 한다.

>> 입력값: 200
출력값: 180
>> 입력값: 15
출력값: 0

In [39]:
num = input()
num = int(num) - 20
if num < 0 :
    print(0)
elif num > 255 :
    print(255)
else :
    print(int(num))

19
0


In [41]:
num = input()
num = int(num) - 20
if num < 0 :
    print(0)
elif num > 255 :
    print(255)
else :
    print(int(num))

286
255


In [42]:
num = input()
num = int(num) - 20
if num < 0 :
    print(0)
elif num > 255 :
    print(255)
else :
    print(int(num))

12
0


116
사용자로부터 입력 받은 시간이 정각인지 판별하라.

>> 현재시간:02:00
정각 입니다.
>> 현재시간:03:10
정각이 아닙니다

In [60]:
print('시간을 입력해주세요')
hour = input()
print('분을 입력해주세요')
minute = int(input())
if minute > 0 :
    print('정각이 아닙니다.')
else : 
    print('정각 입니다.')

시간을 입력해주세요
23
분을 입력해주세요
0
정각 입니다.


In [87]:
print('시간을 형식에 맞춰 입력해주세요(hh:mm)')
time = input()
if int(time[-2:]) > 0 :
    print('정각이 아닙니다.')
else :
    print('정각 입니다.')

시간을 형식에 맞춰 입력해주세요(hh:mm)
23:00
정각 입니다.


In [88]:
time = input()
type(time) # 숫자 입력 했을때

33


str

117
사용자로 입력받은 단어가 아래 fruit 리스트에 포함되어 있는지를 확인하라. 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = ["사과", "포도", "홍시"]
>> 좋아하는 과일은? 사과
정답입니다.

In [93]:
fruit = ["사과", "포도", "홍시"]
print('좋아하는 과일은?')
fav = input()
if fav in fruit :
    print('정답입니다.')
else :
    print('오답입니다.')

좋아하는 과일은?
바나나
오답입니다.


119
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 키 (key) 값에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
>> 제가좋아하는계절은: 봄
정답입니다.

In [104]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

print('제가좋아하는계절은:')
fav = input()
if fav in fruit.keys() :
    print('정답입니다.')
else :
    print('오답입니다.')

제가좋아하는계절은:
겨울
오답입니다.


120
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 값 (value)에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
>> 좋아하는과일은? 한라봉
오답입니다.

In [107]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

print('제가좋아하는과일은:')
fav = input()
if fav in fruit.values() :
    print('정답입니다.')
else :
    print('오답입니다.')

제가좋아하는과일은:
사과
정답입니다.


121
사용자로부터 문자 한 개를 입력 받고, 소문자일 경우 대문자로, 대문자 일 경우, 소문자로 변경해서 출력하라.

>> a
A
힌트-1 : islower() 함수는 문자의 소문자 여부를 판별합니다. 만약 소문자일 경우 True, 대문자일 경우 False를 반환합니다. 힌트-2 : upper() 함수는 대문자로, lower() 함수는 소문자로 변경합니다.

In [111]:
print('대, 소문자 중 아무거나 문자열 1개를 입력해주세요. \n입력하신 값이 소문자일 경우 대문자로, 대문자일 경우, \n소문자로 변경해서 출력해드립니다.')
str = input()
if str.islower() :
    print(str.upper())
else :
    print(str.lower())

대, 소문자 중 아무거나 문자열 1개를 입력해주세요. 
입력하신 값이 소문자일 경우 대문자로, 대문자일 경우, 
소문자로 변경해서 출력해드립니다.
B
b


122
점수 구간에 해당하는 학점이 아래와 같이 정의되어 있다. 사용자로부터 score를 입력받아 학점을 출력하라.

점수	학점
81~100	A
61~80	B
41~60	C
21~40	D
0~20	E
>> score: 83
grade is A

In [117]:
print('점수를 입력해주세요')
score = int(input())
if score > 80 :
    print('grade is A')
elif score > 60 :
    print('grade is B')
elif score > 40 :
    print('grade is C')
elif score > 20 :
    print('grade is D')
else :
    print('grade is E')

점수를 입력해주세요
43
grade is C


123
사용자로부터 달러, 엔, 유로, 또는 위안 금액을 입력받은 후 이를 원으로 변환하는 프로그램을 작성하라. 각 통화별 환율은 다음과 같다. 사용자는 100 달러, 1000 엔, 13 유로, 100 위안과 같이 금액과 통화명 사이에 공백을 넣어 입력한다고 가정한다.

통화명	환율
달러	1167
엔	1.096
유로	1268
위안	171

In [140]:
환율 = {'달러' : 1167, '엔' : 1.096, '유로' : 1268, '위안' : 171}
print('환전할 종류를 입력해주세요 - 달러, 엔, 유로, 위안 중에 입력해주세요')
a = input()
print('환전할 금액을 입력해주세요')
b = int(input())
print(환율[a]*b, '원으로 환전해드리겠습니다.')

환전할 종류를 입력해주세요 - 달러, 엔, 유로, 위안 중에 입력해주세요
달러
환전할 금액을 입력해주세요
134
156378 원으로 환전해드리겠습니다.


124
사용자로부터 세 개의 숫자를 입력 받은 후 가장 큰 숫자를 출력하라.

>> input number1: 10
>> input number2: 9
>> input number3: 20
20

In [146]:
max?

In [152]:
print('첫번째 숫자를 입력해주세요')
num1 = input()
print('두번째 숫자를 입력해주세요')
num2 = input()
print('세번째 숫자를 입력해주세요')
num3 = input()

list1 = [num1, num2, num3]


print('입력하신 값 중 가장 큰 값은', max(list1), '입니다.')

#num1 = input("input number1: ")
#num2 = input("input number2: ")
#num3 = input("input number3: ")
#num1 = int(num1)
#num2 = int(num2)
#num3 = int(num3)

#if num1 >= num2 and num1 >= num3:
#    print(num1)
#elif num2 >= num1 and num2 >= num3:
#    print(num2)
#else:
#    print(num3)
#    ###

첫번째 숫자를 입력해주세요
4
두번째 숫자를 입력해주세요
3
세번째 숫자를 입력해주세요
2
입력하신 값 중 가장 큰 값은 4 입니다.


125
휴대폰 번호 앞자리에 따라 통신사는 아래와 같이 구분된다. 사용자로부터 휴대전화 번호를 입력 받고, 통신사를 출력하는 프로그램을 작성하라.

번호	통신사
011	SKT
016	KT
019	LGU
010	알수없음

In [165]:
'휴대폰 번호를 "000-0000-0000" 형식으로 입력해주세요'
phone = input()
list1 = phone.split('-')
list1

if list1[0] == '011' :
    print('사용하시는 통신사는 SKT 입니다.')
elif list[0] == '016' :
    print('사용하시는 통신사는 KT 입니다.')
elif list[0] == '019' :
    print('사용하시는 통신사는 LGU 입니다.')
else :
    print('사용하시는 통신사를 알 수 없습니다.')

010-2582-1613
사용하시는 통신사를 알 수 없습니다.


126
우편번호는 5자리로 구성되는데, 앞의 세자리는 구를 나타낸다. 예를들어, 강북구의 경우 010, 011, 012 세 자리로 시작한다.

-	0	1	2	3	4	5	6	7	8	9
01	강북구	강북구	강북구	도봉구	도봉구	도봉구	노원구	노원구	노원구	노원구
사용자로 부터 5자리 우편번호를 입력받고 구를 판별하라

>> 우편번호: 01400
도봉구

In [174]:
print('우편번호 5자리를 입력해주세요')

addr = int(input())
if addr < 1300 :
    print('강북구')
elif addr < 1600 :
    print('도봉구')
else :
    print('노원구')
    
    
#우편번호 = input("우편번호: ")
#우편번호 = 우편번호[:3]
#if 우편번호 in ["010", "011", "012"]:
#    print("강북구")
#elif 우편번호 in ["014", "015", "016"]:
#    print("도봉구")
#else:
#    print("노원구")

우편번호 5자리를 입력해주세요
01256
강북구


127
주민등록번호 뒷 자리 7자리 중 첫째 자리는 성별을 나타내는데, 1, 3은 남자 2, 4는 여자를 의미한다. 사용자로부터 13자리의 주민등록번호를 입력 받은 후 성별 (남자, 여자)를 출력하는 프로그램을 작성하라.

>> 주민등록번호: 821010-1635210
남자

In [183]:
print('주민등록번호를 000000-0000000 형태로 입력해주세요')
a = input()
if a[7] == '1' or a[7] == '3' :
    print('남자')
else :
    print('여자')

주민등록번호를 000000-0000000 형태로 입력해주세요
123456-2345678
여자


128
주민등록번호의 뒷 자리 7자리 중 두번째와 세번째는 지역코드를 의미한다. 주민 등록 번호를 입력 받은 후 출생지가 서울인지 아닌지 판단하는 코드를 작성하라

지역코드	출생지
00 ~ 08	서울
09 ~ 12	부산
>> 주민등록번호: 821010-1635210
서울이 아닙니다.
>> 주민등록번호: 861010-1015210
서울 입니다.

In [192]:
print('주민등록번호를 000000-0000000 형태로 입력해주세요')
a = input()
if int(a[8:10]) < 9 :
    print('서울입니다.')
else :
    print('서울이 아닙니다.')

주민등록번호를 000000-0000000 형태로 입력해주세요
123456-1234567
서울이 아닙니다.


129
주민등록번호는 13자리로 구성되는데 마지막 자리수는 주민등록번호의 유효성을 체크하는데 사용된다. 먼저 앞에서부터 12자리의 숫자에 2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5를 차례로 곱한 뒤 그 값을 전부 더한다. 연산 결과 값을 11로 나누면 나머지가 나오는데 11에서 나머지를 뺀 값이 주민등록번호의 마지막 번호가 된다.

  8 2 1 0 1 0 - 1 6 3 5 2 1 0
x 2 3 4 5 6 7   8 9 2 3 4 5 
-----------------------------
1차 계산: (8*2 + 2*3 + 1*4 + 0*5 + 1*6 + 0*7 + 1*8 + 6*9 + 3*2 + 5*3 + 2*4 + 1*5) = (128 % 11) = 7
2차 계산: 11 -7 = 4
위와 같이 821010-1635210에 대해서 계산을 해보면 마지막 자리는 4가 되어야 함을 알 수 있다. 즉, 821010-1635210은 유효하지 않은 주민등록번호임을 알 수 있다.

다음과 같이 사용자로부터 주민등록번호를 입력받은 후 주민등록번호가 유효한지를 출력하는 프로그램을 작성하라.

>> 주민등록번호: 821010-1635210
유효하지 않은 주민등록번호입니다. 

In [247]:
a = input('주민등록번호를 000000-0000000 형태로 입력해주세요 : ')
# b = [2, 3, 4, 5, 6, 7, 0, 8, 9, 2, 3, 4, 5]
c = int(a[0]) * 2 + int(a[1]) * 3 + int(a[2]) * 4 + int(a[3]) * 5 + int(a[4]) * 6 + int(a[5]) * 7 + int(a[7]) * 8 + int(a[8]) * 9 + int(a[9]) * 2 + int(a[10]) * 3 + int(a[11]) * 4 + int(a[12]) * 5 
d = c % 11
e = 11 - d

if a[:-1] == e :
    print('유효한 주민등록번호 입니다.')
else :
    print('유효하지 않은 주민등록번호 입니다.')

주민등록번호를 000000-0000000 형태로 입력해주세요 : 123456-1234567
유효하지 않은 주민등록번호 입니다.


130
아래 코드는 비트코인의 가격 정보를 딕셔너리로 가져오는 코드이다.

import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']
btc 딕셔너리 안에는 시가, 종가, 최고가, 최저가 등이 저장되어 있다. 최고가와 최저가의 차이를 변동폭으로 정의할 때 (시가 + 변동폭)이 최고가 보다 높을 경우 "상승장", 그렇지 않은 경우 "하락장" 문자열을 출력하라.

Key Name	Description
opening_price	최근 24시간 내 시작 거래금액
closing_price	최근 24시간 내 마지막 거래금액
min_price	최근 24시간 내 최저 거래금액
max_price	최근 24시간 내 최고 거래금액

In [248]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']
btc

{'opening_price': '49547000',
 'closing_price': '50127000',
 'min_price': '49500000',
 'max_price': '50330000',
 'units_traded': '3825.90937961',
 'acc_trade_value': '190822958948.1412',
 'prev_closing_price': '49536000',
 'units_traded_24H': '3837.9357942',
 'acc_trade_value_24H': '191418370993.513',
 'fluctate_24H': '634000',
 'fluctate_rate_24H': '1.28',
 'date': '1701183217210'}

In [259]:
최고가 = int(btc['max_price'])
최고가
최저가 = int(btc['min_price'])
최저가
시가 = int(btc['opening_price'])
시가
변동폭 = 최고가 - 최저가
변동폭

if (시가 + 변동폭) > 최고가 :
    print('상승장')
else :
    print('하락장')

상승장


131
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print(변수)

In [ ]:
사과
귤
수박


132
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
  print("#####")

In [ ]:
#####
#####
#####

133
다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.

for 변수 in ["A", "B", "C"]:
  print(변수)

In [3]:
a = ['A', 'B', 'C']

for i in a:
    print(i)
    
# print('A')
# print('B')
# print('C')

a
b
c


134
for문을 풀어서 동일한 동작을하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

In [ ]:
print("출력:", 'A')
print("출력:", 'B')
print("출력:", 'C')

135
for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  b = 변수.lower()
  print("변환:", b)

In [ ]:
print("변환:", 'a')
print("변환:", 'b')
print("변환:", 'c')

136
다음 코드를 for문으로 작성하라.

변수 = 10
print(변수)
변수 = 20
print(변수)
변수 = 30
print(변수)

In [5]:
for i in [10, 20, 30] :
    print(i)

10
20
30


137
다음 코드를 for문으로 작성하라.

print(10)
print(20)
print(30)

In [ ]:
for i in [10, 20, 30] :
    print(i)

138
다음 코드를 for문으로 작성하라.

print(10)
print("-------")
print(20)
print("-------")
print(30)
print("-------")

In [6]:
for i in [10, 20, 30] :
    print(i)
    print("-------")

10
-------
20
-------
30
-------


139
다음 코드를 for문으로 작성하라.

print("++++")
print(10)
print(20)
print(30)

In [7]:
print("++++")
for i in [10, 20, 30] :
    print(i)

++++
10
20
30


140
다음 코드를 for문으로 작성하라.

print("-------")
print("-------")
print("-------")
print("-------")

In [8]:
for i in [1, 2, 3, 4] :
    print("-------")

-------
-------
-------
-------


141
다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

리스트 = [100, 200, 300]
110
210
310

In [9]:
리스트 = [100, 200, 300]
for i in 리스트 :
    print(i + 10 )

110
210
310


142
for 문을 사용해서 리스트에 저장된 값을 다음과 같이 출력하라.

리스트 = ["김밥", "라면", "튀김"]
오늘의 메뉴: 김밥
오늘의 메뉴: 라면
오늘의 메뉴: 튀김

In [10]:
리스트 = ['김밥', '라면', '튀김']
for i in 리스트 :
    print('오늘의 메뉴:', i)

오늘의 메뉴: 김밥
오늘의 메뉴: 라면
오늘의 메뉴: 튀김


143
리스트에 주식 종목이름이 저장돼 있다.

리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
저장된 문자열의 길이를 다음과 같이 출력하라.

6
4
4

In [26]:
리스트 = ["SK하이닉스", "삼성전자", "LG전자"]

for i in 리스트 :
    j = len(i)
    print(j)

6
4
4


144
리스트에는 동물이름이 문자열로 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']
동물 이름과 글자수를 다음과 같이 출력하라.

dog 3
cat 3
parrot 6

In [27]:
리스트 = ['dog', 'cat', 'parrot']

for i in 리스트 :
    j = len(i)
    print(i, j)

dog 3
cat 3
parrot 6


145
리스트에 동물 이름 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']
for문을 사용해서 동물 이름의 첫 글자만 출력하라.

d
c
p

In [57]:
리스트 = ['dog', 'cat', 'parrot']

for i in 리스트 :
 #   j = 리스트[i]
    print(i[0])

d
c
p


146
리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]
for문을 사용해서 다음과 같이 출력하라.

3 x 1
3 x 2
3 x 3

In [40]:
리스트 = [1, 2, 3]

for i in 리스트 :
    print('3 x', i)

3 x 1
3 x 2
3 x 3


147
리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]
for문을 사용해서 다음과 같이 출력하라.

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9

In [48]:
리스트 = [1, 2, 3]

for i in 리스트 :
    j = (3, 'x', i)
j
    

(3, 'x', 3)

148
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

나
다
라

In [69]:
리스트 = ["가", "나", "다", "라"]
리스트 = 리스트[1:]
for i in 리스트 :
    print(i)

나
다
라


150
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

라
다
나
가

In [78]:
리스트 = ["가", "나", "다", "라"]
리스트.reverse()
리스트

for i in 리스트 :
    print(i)
    
##

라
다
나
가


151
리스트에는 네 개의 정수가 저장돼 있다.

리스트 = [3, -20, -3, 44]
for문을 사용해서 리스트의 음수를 출력하라.

-20
-3

In [80]:
리스트 = [3, -20, -3, 44]
result = [i for i in 리스트 if i < 0]
result


for i in 리스트 :
    if i < 0 :
        print(i)
    

-20
-3


152
for문을 사용해서 3의 배수만을 출력하라.

리스트 = [3, 100, 23, 44]
3

In [82]:
리스트 = [3, 100, 23, 44]

for i in 리스트 :
    if i % 3 == 0 :
        print(i)

3


153
리스트에서 20 보다 작은 3의 배수를 출력하라

리스트 = [13, 21, 12, 14, 30, 18]
12
18

In [84]:
리스트 = [13, 21, 12, 14, 30, 18]
for i in 리스트 :
    if i < 20 and i % 3 == 0 :
        print(i)

12
18


154
리스트에서 세 글자 이상의 문자를 화면에 출력하라

리스트 = ["I", "study", "python", "language", "!"]
study
python
language

In [88]:
리스트 = ["I", "study", "python", "language", "!"]

for i in 리스트 :
    j = len(i)
    if j >= 3 :
        print(i)
    

study
python
language


155
리스트에서 대문자만 화면에 출력하라.

리스트 = ["A", "b", "c", "D"]
A
D

In [90]:
리스트 = ["A", "b", "c", "D"]

for i in 리스트 :
    if i.isupper() :
        print(i)

A
D


156
리스트에서 소문자만 화면에 출력하라.

리스트 = ["A", "b", "c", "D"]
b
c

In [91]:
리스트 = ["A", "b", "c", "D"]

for i in 리스트 :
    if i.islower() :
        print(i)

b
c


157
이름의 첫 글자를 대문자로 변경해서 출력하라.

리스트 = ['dog', 'cat', 'parrot']
Dog
Cat
Parrot

In [95]:
리스트 = ['dog', 'cat', 'parrot']

for i in 리스트 :
    j = i[0]
    j = j.upper()
    print(j + i[1:])

Dog
Cat
Parrot


158
파일 이름이 저장된 리스트에서 확장자를 제거하고 파일 이름만 화면에 출력하라. (힌트: split() 메서드)

리스트 = ['hello.py', 'ex01.py', 'intro.hwp']
hello
ex01
intro

In [122]:
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']

for i in 리스트 :
    j = i.split('.')
    k = j[0]
    print(k)


hello
ex01
intro


159
파일 이름이 저장된 리스트에서 확장자가 .h인 파일 이름을 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
intra.h
define.h

In [129]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']


for i in 리스트 :
    j = i.split('.')
    k = j[1]
    if k == 'h' :
        print(i)

intra.h
define.h


160
파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
intra.h
intra.c
define.h

In [130]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']

for i in 리스트 :
    j = i.split('.')
    k = j[1]
    if k == 'h' or k == 'c' :
        print(i)

intra.h
intra.c
define.h


161
for문과 range 구문을 사용해서 0~99까지 한 라인에 하나씩 순차적으로 출력하는 프로그램을 작성하라.

In [137]:
for i in range(100) :
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


162
월드컵은 4년에 한 번 개최된다. range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.

2002
2006
2010
...
2042
2046
2050

In [147]:
for i in range(2002, 2051) : 
    if i % 4 == 2 :
        print(i)
        
        
# for x in range(2022, 2051, 4) :
#   print(x)

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046
2050


163
1부터 30까지의 숫자 중 3의 배수를 출력하라.

3 
6 
9 
12 
15 
18 
21 
24 
27 
30

In [148]:
for i in range(3, 31, 3) :
    print(i)

3
6
9
12
15
18
21
24
27
30


164
99부터 0까지 1씩 감소하는 숫자들을, 한 라인에 하나씩 출력하라.

In [153]:
for i in range(100) :
    print(99-i)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


165
for문을 사용해서 아래와 같이 출력하라.

0.0
0.1
0.2
0.3
0.4
0.5
...
0.9


In [163]:
for i in range(10) :
    print(round(0.1 * i, 1))
    
    
# for num in range(10) : 
#    print(num / 10)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


166
구구단 3단을 출력하라.

3x1 = 3
3x2 = 6
3x3 = 9
3x4 = 12
3x5 = 15
3x6 = 18
3x7 = 21
3x8 = 24
3x9 = 27

In [173]:
for i in range(1,10) :
    print(f'3x{i} =', 3*i)

3x1 = 3
3x2 = 6
3x3 = 9
3x4 = 12
3x5 = 15
3x6 = 18
3x7 = 21
3x8 = 24
3x9 = 27


167
구구단 3단을 출력하라. 단 홀수 번째만 출력한다.

3x1 = 3
3x3 = 9
3x5 = 15
3x7 = 21
3x9 = 27

In [172]:
for i in range(10) :
    if i % 2 == 1 :
        print(f'3x{i} =', 3*i)

3x1 = 3
3x3 = 9
3x5 = 15
3x7 = 21
3x9 = 27


168
1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

합 : 55

In [176]:
합 = 0
for i in range(1, 11) :
    합 = 합 + i
print('합 :', 합)

합 : 55


169
1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.

합: 25

In [180]:
합 = 0
for i in range(1, 11) :
    if i % 2 == 1 :
        합 = 합 + i
print('합 =', 합)

합 = 25


170
1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [183]:
곱 = 1
    
for i in range(1, 11) :
    곱 = 곱 * i
print('곱 =', 곱)

곱 = 3628800


171
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]
32100
32150
32000
32500

In [192]:
price_list = [32100, 32150, 32000, 32500]

for i in price_list :
    print(i)

32100
32150
32000
32500


172
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]
0 32100
1 32150
2 32000
3 32500

In [193]:
price_list = [32100, 32150, 32000, 32500]

for i, j in enumerate(price_list) :
    print(i, j)
    
    
    ### 이게 ragne 인가?

0 32100
1 32150
2 32000
3 32500


173
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]
3 32100
2 32150
1 32000
0 32500

In [202]:
price_list = [32100, 32150, 32000, 32500]

for i, j in enumerate(price_list) :
    print(3-i, j)
    
# for i in range(len(price_list)) :
#    print(3, -i, price_list[i])

# for i in range(len(price_list)) :
#    print((len(price_list) - 1) - i, price_list[i])

3 32100
2 32150
1 32000
0 32500


174
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]
100 32150
110 32000
120 32500

In [209]:
price_list = [32100, 32150, 32000, 32500]

for i in range(1, len(price_list)) :
    print(100 + i*10, price_list[i])

110 32150
120 32000
130 32500


175
my_list를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라"]
가 나
나 다
다 라

In [215]:
my_list = ["가", "나", "다", "라"]

for i in range(len(my_list)-1) :
    print(my_list[i], my_list[i+1])

가 나
나 다
다 라


176
리스트를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라", "마"]
가 나 다
나 다 라
다 라 마

In [219]:
my_list = ["가", "나", "다", "라", "마"]

for i in range(len(my_list)-2) :
    print(my_list[i], my_list[i+1], my_list[i+2])

가 나 다
나 다 라
다 라 마


177
반복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라"]
라 다
다 나
나 가

In [228]:
my_list = ["가", "나", "다", "라"]

for i in range(len(my_list)-1) :
    print(my_list[len(my_list) -1 -i], my_list[len(my_list) -2 -i])

라 다
다 나
나 가


178
리스트에는 네 개의 정수가 저장되어 있다. 각각의 데이터에 대해서 자신과 우측값과의 차분값을 화면에 출력하라.

my_list = [100, 200, 400, 800]
예를들어 100을 기준으로 우측에 위치한 200과의 차분 값를 화면에 출력하고, 200을 기준으로 우측에 위치한 400과의 차분값을 화면에 출력한다. 이어서 400을 기준으로 우측에 위치한 800과의 차분값을 화면에 출력한다.

100
200
400

In [230]:
my_list = [100, 200, 400, 800]

for i in range(len(my_list) -1) :
    print(my_list[i+1] - my_list[i])

100
200
400


179
리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.

my_list = [100, 200, 400, 800, 1000, 1300]
첫 번째 줄에는 100, 200, 400의 평균값이 출력된다. 두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력한다.

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333

In [262]:
my_list = [100, 200, 400, 800, 1000, 1300]
sum = 0

for i in range(1, len(my_list) -1) :
    print((my_list[i-1] + my_list[i] + my_list[i+1]) / 3)

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333


180
리스트에 5일간의 저가, 고가 정보가 저장돼 있다. 고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.

low_prices  = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]

In [265]:
low_prices  = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]
volatility = []

for i in range(len(low_prices)) :
    low = low_prices[i]
    high = high_prices[i]
    volatility.append(high - low)
print(volatility)
        
    
## for i in range(len(low_prices)) :
##    volatility.append(hight_price[i] - low_price[i])

[50, 100, 30, 80, 0]


181
아래 표에서 하나의 행을 하나의 리스트로, 총 3개의 리스트를 갖는 이차원 리스트 apart를 정의하라.

101호	102호
201호	202호
301호	302호

In [266]:
apart = [['101호', '102호'], ['201호', '202호'], ['301호', '302호']]
apart

[['101호', '102호'], ['201호', '202호'], ['301호', '302호']]

182
아래 표에서 하나의 열을 하나의 리스트로, 총 2개의 리스트를 갖는 이차원 리스트 stock을 정의하라.

시가	종가
100	80
200	210
300	330

In [268]:
stock = [['시가', 100, 200, 300], ['종가', 80, 210, 330]]
stock

[['시가', 100, 200, 300], ['종가', 80, 210, 330]]

183
아래 표를 stock 이름의 딕셔너리로 표현하라.시가를 key로 저장하고, 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다. 종가 역시 key로 저장하고 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다.

시가	종가
100	80
200	210
300	330

In [269]:
stock = {'시가' : [100, 200, 300], '종가' : [80, 210, 330]}
stock

{'시가': [100, 200, 300], '종가': [80, 210, 330]}

184
아래 표를 stock 이라는 이름의 딕셔너리로 표현하라. 날짜를 key로 저장하고, 나머지 같은 행의 데이터를 리스트로 저장해서 value로 저장한다. 첫 열이 날짜이다.

10/10	80	110	70	90
10/11	210	230	190	200

In [270]:
stock = {'10/10' : [80, 110, 70, 90], '10/11' : [210, 230, 190, 200]}
stock

{'10/10': [80, 110, 70, 90], '10/11': [210, 230, 190, 200]}

185
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
101 호
102 호
201 호
202 호
301 호
302 호

In [274]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in apart :
    for j in i :
        print(j, '호')

101 호
102 호
201 호
202 호
301 호
302 호


186
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
301 호
302 호
201 호
202 호
101 호
102 호

In [291]:
apart = [ [101, 102], [201, 202], [301, 302] ]
apart.reverse()
for i in apart :
    for j in i :
        print(j, '호')
        
#for row in apart[::-1]:
#    for col in row:
#        print(col, "호")

301 호
302 호
201 호
202 호
101 호
102 호


187
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
302 호
301 호
202 호
201 호
102 호
101 호

In [292]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in apart[::-1] :
    for j in i[::-1] :
        print(j, '호')

302 호
301 호
202 호
201 호
102 호
101 호


188
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
101 호
-----
102 호
-----
201 호
-----
202 호
-----
301 호
-----
302 호
-----

In [294]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in apart :
    for j in i :
        print(j, '호')
        print('-----')
        
        
        # print('-' * 5)

101 호
-----
102 호
-----
201 호
-----
202 호
-----
301 호
-----
302 호
-----


189
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
101 호
102 호
-----
201 호
202 호
-----
301 호
302 호
-----


In [295]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in apart :
    for j in i :
        print(j, '호')
    print('-' * 5)

101 호
102 호
-----
201 호
202 호
-----
301 호
302 호
-----


190
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]
101 호
102 호
201 호
202 호
301 호
302 호
-----

In [296]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in apart :
    for j in i :
        print(j, '호')
print('-' * 5)

101 호
102 호
201 호
202 호
301 호
302 호
-----


191
data에는 매수한 종목들의 OHLC (open/high/low/close) 가격 정보가 바인딩 되어있다.

data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

수수료를 0.014 %로 가정할 때, 각 가격에 수수료를 포함한 가격을 한라인에 하나씩 출력하라.

2000.28
3050.427
2050.2870000000003
...

In [316]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

수수료 = 0.00014

for i in data : 
    for j in i :
        value = j + j * 수수료
        print(value)

2000.28
3050.427
2050.287
1980.2772
7501.05
2050.287
2050.287
1980.2772
15452.163
15052.107
15552.177
14902.086


192
191번의 출력 결과에 행단위로 "----" 구분자를 추가하라.

2000.28
3050.427
2050.2870000000003
1980.2772
----
7501.05
2050.2870000000003
2050.2870000000003
1980.2772
----
15452.163
15052.107
15552.177
14902.086000000001
----

In [317]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

수수료 = 0.00014

for i in data : 
    for j in i :
        value = j + j * 수수료
        print(value)
    print('-'*4)

2000.28
3050.427
2050.287
1980.2772
----
7501.05
2050.287
2050.287
1980.2772
----
15452.163
15052.107
15552.177
14902.086
----


193
192 번 문제의 결괏값을 result 이름의 리스트에 1차원 배열로 저장하라.

>> print(result)
[2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, ...]

In [320]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

result = []

수수료 = 0.00014

for i in data : 
    for j in i :
        value = j + j * 수수료
        result.append(value)
print(result) 

[2000.28, 3050.427, 2050.287, 1980.2772, 7501.05, 2050.287, 2050.287, 1980.2772, 15452.163, 15052.107, 15552.177, 14902.086]


194
191번 문제의 결괏값을 result 이름의 리스트에 2차원 배열로 저장하라. 저장 포맷은 아래와 같다. 각 행에 대한 데이터끼리 리스트에 저장되어야 한다.

>> print(result)
[
 [2000.28, 3050.427, 2050.2870000000003, 1980.2772],
 [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772],
 [15452.163, 15052.107, 15552.177, 14902.086000000001]
]

In [327]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

result = []
수수료 = 0.00014

for i in data : 
    list1 = []
    for j in i :
        value = j + j * 수수료
        list1.append(value)
    result.append(list1)
print(result) 

[[2000.28, 3050.427, 2050.287, 1980.2772], [7501.05, 2050.287, 2050.287, 1980.2772], [15452.163, 15052.107, 15552.177, 14902.086]]


195
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 화면에 종가데이터를 출력하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
100
190
310

In [360]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]



for i in ohlc[1:] :
    count = 0
    for j in i :
        count += 1
        if count == 4 :
            print(j)
            
            
#for row in ohlc[1:] :
#    print(row[3])

100
190
310
100
190
310


196
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 150원보다 큰경우에만 종가를 출력하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
190
310

In [361]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
         
           
for row in ohlc[1:] :
    if row[3] > 150 :
        print(row[3])

190
310


197
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 시가 보다 크거나 같은 경우에만 종가를 출력하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
100
310

In [362]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
         
           
for row in ohlc[1:] :
    if row[3] >= row[0] :
        print(row[3])

100
310


198
ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 고가와 저가의 차이를 변동폭으로 정의할 때 변동폭을 volatility 이름의 리스트에 저장하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
>> print(volatility)
[40, 30, 10]

In [366]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
         
volatility = []           
for row in ohlc[1:] :
    volatility.append(row[1] - row[2])
print(volatility)

[40, 30, 10]


199
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 종가가 시가보다 높은 날의 변동성 (고가 - 저가)을 화면에 출력하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
종가가 시가보다 높은 거래일의 OHLC는 [300, 310, 300, 310] 이다. 따라서 이 거래일의 변동성은 10 (310 - 300)이다.

10

In [369]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
         
for row in ohlc[1:] :
    if row[3] > row[0] :
        volatility = row[1] - row[2]
        print(volatility)
        
#for row in ohlc[1:]:
#    if row[3] > row[0]:
#        print(row[1]-row[2])


10


200
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 시가에 매수해서 종가에 매도 했을 경우 총 수익금을 계산하라.

ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
1일차 수익 0원 (100 - 100), 2일차 수익 -10원 (190 - 200), 3일차 수익 10원 (310 - 300) 이다.

0

In [378]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
         
일차 = 0

for row in ohlc[1:] :
    일차 += 1
    시가 = row[0]
    종가 = row[3]
    수익 = row[3] - row[0]
    print(f'{일차}일차 수익 {수익}원 ({종가} - {시가})')

1일차 수익 0원 (100 - 100)
2일차 수익 -10원 (190 - 200)
3일차 수익 10원 (310 - 300)


In [382]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
수익 = 0   
총수익 = 0
for row in ohlc[1:] :
    시가 = row[0]
    종가 = row[3]
    수익 = row[3] - row[0]
    총수익 = 총수익 + 수익
print(총수익)


# profit = 0
# for row in ohlc[1:] :
#    profit += (row[3] - row[0])

0


201
"비트코인" 문자열을 화면에 출력하는 print_coin() 함수를 정의하라.

In [15]:
def print_coin() :
    print('비트코인')

202
201번에서 정의한 함수를 호출하라.

In [16]:
print_coin()

비트코인


203
201번에서 정의한 print_coin 함수를 100번호출하라.

In [17]:
for i in range(100) :
    print_coin()

비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인
비트코인


204
"비트코인" 문자열을 100번 화면에 출력하는 print_coins() 함수를 정의하라.

In [18]:
def print_coin() :
    for i in range(100) :
        print('비트코인')

205
아래의 에러가 발생하는 이유에 대해 설명하라.

hello()
def hello():
    print("Hi")
실행 예

NameError: name 'hello' is not defined

In [ ]:
hello 함수가 정의 되기전에 호출하여서 에러가 발생함

206
아래 코드의 실행 결과를 예측하라.

def message() :
    print("A")
    print("B")

message()
print("C")
message()

In [ ]:
A
B
C
A
B

207
아래 코드의 실행 결과를 예측하라. (읽기 어려운 코드의 예입니다.)

print("A")

def message() :
    print("B")

print("C")
message()

In [ ]:
A
C
B


208
아래 코드의 실행 결과를 예측하라. (읽기 어려운 코드의 예입니다.)

print("A")
def message1() :
    print("B")
print("C")
def message2() :
    print("D")
message1()
print("E")
message2()

In [ ]:
A
C
B
E
D


209
아래 코드의 실행 결과를 예측하라.

def message1():
    print("A")

def message2():
    print("B")
    message1()

message2()

In [ ]:
B
A

210
아래 코드의 실행 결과를 예측하라.

def message1():
    print("A")

def message2():
    print("B")

def message3():
    for i in range (3) :
        message2()
        print("C")
    message1()

message3()

In [ ]:
B
C
B
C
B
C
A

211
함수의 호출 결과를 예측하라.

def 함수(문자열) :
    print(문자열)

함수("안녕")
함수("Hi")

In [ ]:
안녕
Hi

212
함수의 호출 결과를 예측하라.

def 함수(a, b) :
    print(a + b)

함수(3, 4)
함수(7, 8)

In [ ]:
7
15


213
아래와 같은 에러가 발생하는 원인을 설명하라.

def 함수(문자열) :
    print(문자열)
함수()
TypeError: 함수() missing 1 required positional argument: '문자열'

In [ ]:
함수에 문자열 매개변수가 필요한데 인자값이 없어서 에러가 발생함

214
아래와 같은 에러가 발생하는 원인을 설명하라.

def 함수(a, b) :
    print(a + b)

함수("안녕", 3)
TypeError: must be str, not int

In [ ]:
str 타입인 '안녕' 과 int 타입인 3을 더할 수 없어서 에러가 발생함

215
하나의 문자를 입력받아 문자열 끝에 ":D" 스마일 문자열을 이어 붙여 출력하는 print_with_smile 함수를 정의하라.

In [22]:
def print_with_smile(a) :
    x = str(a) + ':D'
    return x

print_with_smile('Hi')


# def print_with_smile (string) :
#     print (string + ":D")

'Hi:D'

216
215에서 정의한 함수를 호출하라. 파라미터는 "안녕하세요"로 입력하라.

In [23]:
print_with_smile('안녕하세요')

'안녕하세요:D'

217
현재 가격을 입력 받아 상한가 (30%)를 출력하는 print_upper_price 함수를 정의하라.

In [24]:
def print_upper_price(a) :
    x = a * 1.3
    return x

218
두 개의 숫자를 입력받아 두 수의 합을 출력하는 print_sum 함수를 정의하라.

In [25]:
def print_sum(a, b) :
    x = a + b
    return x

219
두 개의 숫자를 입력받아 합/차/곱/나눗셈을 출력하는 print_arithmetic_operation 함수를 작성하라.

print_arithmetic_operation(3, 4)
3 + 4 = 7
3 - 4 = -1
3 * 4 = 12
3 / 4 = 0.75

In [27]:
def print_arithmetic_operation(a, b) :
    print(a, '+', b, '=', a+b)
    print(a, '-', b, '=', a-b)
    print(a, '*', b, '=', a*b)
    print(a, '/', b, '=', a/b)

220
세 개의 숫자를 입력받아 가장 큰수를 출력하는 print_max 함수를 정의하라. 단 if 문을 사용해서 수를 비교하라.

In [32]:
def print_max(a, b, c) :
    if a >= b :
        if a >= c :
            print(a)
        else :
            print(c)
    else :
        if a < b :
            if b >= c :
                print(b)
            else :
                print(c) 
                
                
# def print_max(a, b, c) :
#     max_val = 0
#     if a > max_val :
#         max_val = a
#     if b > max_val :
#         max_val = b
#     if c > max_val :
#         max_val = c
#     print(max_val)

221
입력된 문자열을 역순으로 출력하는 print_reverse 함수를 정의하라.

print_reverse("python")
nohtyp

In [42]:
def print_reverse(a) :
    x = a[::-1]
    return x

222
성적 리스트를 입력 받아 평균을 출력하는 print_score 함수를 정의하라.

print_score ([1, 2, 3])
2.0

In [44]:
def print_score(a) :
    x = sum(a)/len(a)
    return x

223
하나의 리스트를 입력받아 짝수만 화면에 출력하는 print_even 함수를 정의하라.

print_even ([1, 3, 2, 10, 12, 11, 15])
2
10
12

In [69]:
def print_even(a) :
    for i in a :
        if i % 2 == 0 :
            print(i)

224
하나의 딕셔너리를 입력받아 딕셔너리의 key 값을 화면에 출력하는 print_keys 함수를 정의하라.

print_keys ({"이름":"김말똥", "나이":30, "성별":0})
이름
나이
성별

In [77]:
def print_keys(a) :
    for i in a.keys() :
        print(i)
    

225
my_dict에는 날짜를 키값으로 OHLC가 리스트로 저장돼 있다.

my_dict = {"10/26" : [100, 130, 100, 100],
           "10/27" : [10, 12, 10, 11]}
my_dict와 날짜 키값을 입력받아 OHLC 리스트를 출력하는 print_value_by_key 함수를 정의하라.

print_value_by_key  (my_dict, "10/26")
[100, 130, 100, 100]

In [9]:
my_dict = {"10/26" : [100, 130, 100, 100],
           "10/27" : [10, 12, 10, 11]}

def print_value_by_key(a, b) :
    print(a[b])


226
입력 문자열을 한 줄에 다섯글자씩 출력하는 print_5xn(string) 함수를 작성하라.

print_5xn("아이엠어보이유알어걸")
아이엠어보
이유알어걸

In [27]:
def print_5xn(a) :
    x = int(len(a) / 5) # 인수가 5의 몇배수인지 계산해서 x 에 저장
    for i in range(x + 1) : # 
        print(a[i * 5 : i * 5 + 5])

227
문자열과 한줄에 출력될 글자 수를 입력을 받아 한 줄에 입력된 글자 수만큼 출력하는 print_mxn(string) 함수를 작성하라.

printmxn("아이엠어보이유알어걸", 3)
아이엠
어보이
유알어
걸

In [32]:
def printmxn(a, b) :
    x = int(len(a) / b) 
    for i in range(x + 1) : # 
        print(a[i * b : i * b + b])

printmxn("아이엠어보이유알어걸", 3)


In [33]:
printmxn("아이엠어보이유알어걸", 3)


아이엠
어보이
유알어
걸


228
연봉을 입력받아 월급을 계산하는 calc_monthly_salary(annual_salary) 함수를 정의하라. 회사는 연봉을 12개월로 나누어 분할 지급하며, 이 때 1원 미만은 버림한다.

calc_monthly_salary(12000000)
1000000

In [1]:
def calc_monthly_salary(a) :
    x = round(a / 12)
    return x

# x = int(a / 12)

229
아래 코드의 실행 결과를 예측하라.

def my_print (a, b) : #1 a, b라는 매개변수를 지닌 my_print 함수 선언
    print("왼쪽:", a) #3 '왼쪽:' 을 출력하고 입력받은 a 인수를 함께 출력
    print("오른쪽:", b) #4 '오른쪽:' 을 출력하고 입력받은 b 인수를 함께 출력

my_print(a=100, b=200) #2 my_print 함수에 인수값 100, 200을 입력

In [ ]:
# 왼쪽 : 100
# 오른쪽 : 200



230
아래 코드의 실행 결과를 예측하라.

def my_print (a, b) :
    print("왼쪽:", a)
    print("오른쪽:", b)

my_print(b=100, a=200)

In [4]:
# 왼쪽 : 200
# 오른쪽 : 100   

231
아래 코드를 실행한 결과를 예상하라.

def n_plus_1 (n) :
    result = n + 1

n_plus_1(3)
print (result)

In [6]:
# n_plus_1 (3) 에서 지역변수 result 에 3 + 1 값을 넣어주지만 리턴되지 않아서 
# 결과가 나오지 않고 마찬가지로 지역변수이기 때문에 선언되지 않은
# result 를 출력할 수 없어 에러가 표시된다

232
문자열 하나를 입력받아 인터넷 주소를 반환하는 make_url 함수를 정의하라.

make_url("naver")
www.naver.com

In [4]:
def make_url(a) : #1 a라는 매개변수를 지닌 make_url 함수 선언
    print('www.%s.com' % a) #3 'www.%s.com' 문자열을 출력하는데
                            #   % 연산자를 사용해 입력받은 인수 a를 대입함
    
make_url('naver') #2 make_url 함수에 인수값 'naver'를 입력

www.naver.com


233
문자열을 입력받아 각 문자들로 구성된 리스트로 반환하는 make_list 함수를 정의하라.

make_list("abcd")
['a', 'b', 'c', 'd']

In [15]:
def make_list(a) :
    x = []
    for i in range(len(a)) :
        x.append(a[i])
    return x


#def make_list (string) :
#    my_list = []
#    for 변수 in string :
#        my_list.append(변수)
#    return my_list

234
숫자로 구성된 하나의 리스트를 입력받아, 짝수들을 추출하여 리스트로 반환하는 pickup_even 함수를 구현하라.

pickup_even([3, 4, 5, 6, 7, 8])
[4, 6, 8]

In [24]:
def pickup_even(a) :
    x = []
    for a in a :
        if a % 2 == 0 :
            x.append(a)
    return x

pickup_even([3, 4, 5, 6, 7, 8])
                   

[4, 6, 8]

235
콤마가 포함된 문자열 숫자를 입력받아 정수로 변환하는 convert_int 함수를 정의하라.

convert_int("1,234,567")
1234567

In [7]:
def convert_int(a) : #1 a라는 매개변수를 지닌 convert_int 함수 선언
    print(a.replace(',', '')) #3 replace 메소드를 사용해
                              #  입력받은 인수 a에서 ','을 ''로 변환해줌

convert_int('1,234,567') #2 convert_int 함수에 인수값 '1,234,567' 을 입력

1234567


236
아래 코드의 실행 결과를 예측하라.

def 함수(num) :
    return num + 4

a = 함수(10)
b = 함수(a)
c = 함수(b)
print(c)

In [ ]:
# 22

237
아래 코드의 실행 결과를 예측하라.

def 함수(num) :
    return num + 4

c = 함수(함수(함수(10)))
print(c)

In [ ]:
# 22

238
아래 코드의 실행 결과를 예측하라.

def 함수1(num) : #1 num이라는 매개변수를 지닌 함수1을 선언
    return num + 4 #4 입력 받은 인수값 10에 4를 더해서 리턴해줌

def 함수2(num) : #2 num이라는 매개변수를 지닌 함수2를 선언
    return num * 10 #6 입력 받은 인수값 14에 10을 곱해서 리턴해줌

a = 함수1(10) #3 함수1에 인수값 10을 입력해서 리턴 받은 값을 a에 저장
c = 함수2(a) #5 함수2에 인수값 a을 입력. 리턴받은 14(10 + 4)을 입력해서
             #  리턴 받은 값을 c에 저장
print(c) #7 변수 c에 저장된 값 140(14 * 10) 을 출력

In [ ]:
# 140

239
아래 코드의 실행 결과를 예측하라.

def 함수1(num) :
    return num + 4

def 함수2(num) :
    num = num + 2
    return 함수1(num)

c = 함수2(10)
print(c)

In [ ]:
# 16

240
아래 코드의 실행 결과를 예측하라.

def 함수0(num) :
    return num * 2

def 함수1(num) :
    return 함수0(num + 2)

def 함수2(num) :
    num = num + 10
    return 함수1(num)

c = 함수2(2)
print(c)

In [ ]:
# 28